lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -1,7 +1,28 @@
 module ActiveRecord::Import::PostgreSQLAdapter
   include ActiveRecord::Import::ImportSupport
 
+  def insert_many( sql, values, *args ) # :nodoc:
+    number_of_inserts = 1
+
+    base_sql,post_sql = if sql.is_a?( String )
+      [ sql, '' ]
+    elsif sql.is_a?( Array )
+      [ sql.shift, sql.join( ' ' ) ]
+    end
+
+    sql2insert = base_sql + values.join( ',' ) + post_sql
+    ids = select_values( sql2insert, *args )
+
+    [number_of_inserts,ids]
+  end
+
+
   def next_value_for_sequence(sequence_name)
     %{nextval('#{sequence_name}')}
   end
+
+  def supports_setting_primary_key?
```

Comment:
Its really "does the adapter support setting the ids on the models from the bulk import".  For the postgresql adapter, we modify the post_sql_statements to tack on a ```RETURNING #{options[:primary_key]}```.  Maybe the option needs a better name so our future-selves will not want to kick our past selves